src/Modules/AuthModule/components/Register/Register.tsx

```diff
@@ -1,5 +1,403 @@
+                                      value: /[A-Za-z]{4,7}[\d]{1}/gm,
+                                      message:
+                                        "The user name must contain characters and end with numbers without spaces.",
+                                    },
+                                    maxLength: {
+                                      value: 8,
+                                      message:
+                                        "The user name may not be greater than 8 characters.",
+                                    },
+                                    minLength: {
+                                      value: 4,
+                                      message:
+                                        "The user name must be at least 4 characters.",
+                                    },
+                                  })}
+                                />
+                                <label
+                                  htmlFor="input-field"
+                                  className="input-label"
+                                >
+                                  User Name
+                                </label>
+                                <span className="input-highlight"></span>
+                              </div>
+                              <ErrorMessage
+                                errors={errors}
+                                name="userName"
+                                render={({ messages }) => {
+                                  return messages
+                                    ? Object.entries(messages).map(
+                                        ([type, message]) => (
+                                          <p
+                                            className="text-start text-danger ps-2"
+                                            key={type}
+                                          >
+                                            {message}
+                                          </p>
+                                        )
+                                      )
+                                    : null;
+                                }}
+                              />
+
+                              {/* phone input */}
+                              <div className="input-container">
+                                <input
+                                  placeholder="Enter your phone"
+                                  className={`input-field ${
+                                    errors.phoneNumber && "border-danger"
+                                  }`}
+                                  type="text"
+                                  {...register("phoneNumber", {
+                                    required: "Phone number is required",
+                                    pattern: {
+                                      value: /^[\d]{1,18}$/gm,
+                                      message: "Phone digits only",
```

Comment:
Phone numbers can contain  + - .  If we are targeting people in Egypt, we can make the message invalid Egyptian phone number, as numbers in different countries have different formats.